azurerm/internal/services/keyvault/key_vault_secrets_data_source.go

```diff
@@ -0,0 +1,103 @@
+package keyvault
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault/parse"
+	keyVaultValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceKeyVaultSecrets() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceKeyVaultSecretsRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"key_vault_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: keyVaultValidate.VaultID,
+			},
+
+			"max_results": {
+				Type:     pluginsdk.TypeInt,
+				Optional: true,
+				Default:  25,
```

Comment:
This one is API default, hence the number. Happy to raise this all the way to INTMAX, but thought I'd go with what SDK told me.

Reply:
i think it would be better to maybe default to returning them all and then this can optionally be used to reduce the number returned - i can see someone not realizing this property exists and then spending lots of time debugging why they ar eonly getting 25 items returned